framework/src/modules/token/method.ts

```diff
@@ -734,4 +799,28 @@ export class TokenMethod extends BaseMethod {
 			.get(EscrowStore)
 			.has(methodContext, this.stores.get(EscrowStore).getKey(chainID, tokenID));
 	}
+
+	private async _isAllTokensSupported(methodContext: MethodContext): Promise<boolean> {
+		const supportedTokensStore = this.stores.get(SupportedTokensStore);
+
+		return supportedTokensStore.has(methodContext, ALL_SUPPORTED_TOKENS_KEY);
+	}
+
+	private async _getSupportedTokens(
+		methodContext: MethodContext,
+		chainID: Buffer,
+	): Promise<SupportedTokensStoreData> {
+		if (chainID.length === 0 || !chainID) {
+			throw new Error('Chain ID is required.');
+		}
```

Comment:
why is this check necessary? by typescript, chainID is never undefined, and if length needs to be checked, shouldnt we check if the chainID has 4bytes?